### DISCUSSION TITLE
<podcast:pinned> - For "pinning" an episode to the top of a feed

### DISCUSSION THREAD

**Discussion by daveajones** [Podcastindex-org/podcast-namespace#570] [Tag Proposal]
# Proposal:  \<podcast:pinned\> tag

<br>

## Purpose
Sometimes a podcaster may feel that a particular episode (or episodes) are a good entry point into their show and would like to suggest that listeners start with that particular episode first.  The `<podcast:pinned>` tag allows the podcaster to specify one or more such episodes to be "pinned" to the top of the episode list for purposes of display in a podcast app after subscribing.  This "pinning" effect is only meant to be done when a podcast is first subscribed to.  Once the show is listened to or marked as listened, it should no longer be pinned.

## Tag
`<podcast:pinned>`<br><br>

### Parent
&nbsp; `<channel>`
<br>

### Count
&nbsp; Multiple
<br>

### Attributes
- **guid** (required) The `guid` of the episode to be pinned.
<br>

### Examples
Example (IRC):
```xml
<podcast:pinned guid="208a035a-1938-4343-8e1f-f686a8008e23" />
```

<br>

**Comment by daveajones** [comment]
The first thought was having this in the `<channel>` with a `guid` attribute attached.  But, I could also see it just being a child of the pinned `<item>`.  I don't have a strong preference either way.  Maybe one of the hosting companies have a preference?

**Reply by genebean** [reply]
My $0.02: I like it/them being at the channel level because it would be easy to locate without having to iterate over every item in the feed (both programmatically and as a human).

**Reply by ryan-lp** [reply]
Ii would also make a pinned item / featured item / spotlight item as a child of the channel, whereas playlist membership could be tagged via a child of the item. i.e. If we're recreating what YouTube has, it shows a single spotlight video at the top of a channel, and then below it, it shows some featured playlists.

**Comment by theDanielJLewis** [comment]
Relevant prior discussions: #305, #281, and #385.

**Comment by samsethi** [comment]
Hi Dave

Should that be feedGUID for Pinning to ensure consistency? 

<podcast:pinned **feedGuid**="208a035a-1938-4343-8e1f-f686a8008e23" />

            <podcast:valueTimeSplit startTime="60" duration="237" remotePercentage="95">
               <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#1" **feedGuid**="a94f5cc9-8c58-55fc-91fe-a324087a655b" medium="music" />
            </podcast:valueTimeSplit>

<podcast:podroll>
    <podcast:remoteItem **feedGuid**="29cdca4a-32d8-56ba-b48b-09a011c5daa9" />
    <podcast:remoteItem **feedGuid**="396d9ae0-da7e-5557-b894-b606231fa3ea" />
    <podcast:remoteItem **feedGuid**="917393e3-1b1e-5cef-ace4-edaa54e1f810" />
</podcast:podroll>

**Reply by jamescridland** [reply]
"Pinned" is pointing to an episode in the current feed - so, no.

But, it would help if it were called `itemGuid` rather than the ambiguous `guid`.

**Reply by samsethi** [reply]
Yep sorry you are right James itemGUID

**Comment by theDanielJLewis** [comment]
I think I said something similar to this in the prior discussions about a "feature" option, but here's a more refined idea in this latest version.

I suggest we offer to ways to feature an episode: 

1. Wrap a normal `<item>` like we do for live items. This would allow the podcaster to create a normal episode but prevent it from downloading. This could prevent it from downloading on first follow. But it would also prevent it if a podcaster adds a featured item later. For example, does my entire existing audience really need to receive the trailer designed for non-listeners?
2. Add a simple _in-item_ tag for featuring something that _is_ supposed to be released to existing followers. For example, a special episode, announcement, best-of, or anything like that.

I also suggest we offer a `description` attribute, allowing the podcaster to write something specifically targeted to new audiences. Featuring a normal episode, for example, might say, "This episode contains our listener-favorite stories from the past 5 years, and it's a great way to hear our best highlights!" But the actual episode's description would be a normal one.

**Reply by jamescridland** [reply]
This isn't a trailer.

This is an equivalent of Spotify's "START HERE" feature - a publicly available episode that you'd really like people to try first.

I'd be really keen to make it as functionally equivalent to Spotify's feature as possible, please.

**Reply by theDanielJLewis** [reply]
Right, I referenced a trailer as one example where the podcaster might not want the media pushed down the feed. Other instances could be a "best of" compilation, or any kind of "start here" sort of media that may or may not also be a "trailer" of some sort.

**Reply by jamescridland** [reply]
When Spotify launches their "start here" function, I think *that* is the time when we specify this feature.

My understanding is that it's *always* a public episode - even a trailer - and should always exist in an `<item>` tag to be playable.

**Comment by samsethi** [comment]
We added support for Podcast Pinning this week and at the top of a podcaster's feed, they can now either feature their trailer or a previous episode. They do this by simply ticking either the trailer episode which is selected by default or they choose an episode they want NEW listeners to start listening to first before they subscribe to the podcast. 

The Podfans App front end. Showing a trailer episode 
<img width="1326" alt="Podfans___Shows___The_Digital_Her_Podcast" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/1a42f735-07ae-426f-b08f-09fa34ea30a3">


Creators Admin Dashboard showing the trailer episode selected which is featured. 

<img width="1344" alt="Podfans___Dashboard" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/74e62f7f-609b-4e4d-a0be-8de8c859031d">

**Comment by samsethi** [comment]
@Dave This could be added to phase 7 as it's simple and has been implemented to show it working in TrueFans.

**Comment by mijustin** [comment]
@daveajones @samsethi I would like to +1 something like this proposal.

Here is the use-case: we host a popular podcast that has some “greatest hits” episodes that many new listeners don’t see because in an episodic feed, those episodes are waaaaay down the list.

Currently, they’re considering changing the publish date on those episodes so they appear in slots 3-5 (after their newest 2). I don’t love this hack - wish there was a way to give creators this control.

It would be amazing for creators to (metaphorically) welcome new followers to their feed: "Welcome! You'll find our newest episodes here, but also here are our "greatest hits.""

h/t @nathangathright for pointing me here 👍

**Reply by Marzal** [reply]
What about a more broad implementation?

https://github.com/Podcastindex-org/podcast-namespace/discussions/747
